IncomingCallRouting/IncomingCallRouting/Utils/IncomingCallHandler.cs

```diff
@@ -0,0 +1,346 @@
+            var dtmfReceivedEvent = new NotificationCallback((CallingServerEventBase callEvent) =>
+            {
+                Task.Run(async () =>
+                {
+                    var toneReceivedEvent = (ToneReceivedEvent)callEvent;
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"Tone received ---------> : {toneReceivedEvent.ToneInfo?.Tone}");
+
+                    if (toneReceivedEvent?.ToneInfo?.Tone == ToneValue.Tone1)
+                    {
+                        toneReceivedCompleteTask.TrySetResult(true);
+                    }
+                    else
+                    {
+                        toneReceivedCompleteTask.TrySetResult(false);
+                    }
+
+                    EventDispatcher.Instance.Unsubscribe(CallingServerEventType.ToneReceivedEvent.ToString(), callConnectionId);
+                    // cancel playing audio
+                    await CancelAllMediaOperations().ConfigureAwait(false);
+                });
+            });
+            //Subscribe to event
+            EventDispatcher.Instance.Subscribe(CallingServerEventType.ToneReceivedEvent.ToString(), callConnectionId, dtmfReceivedEvent);
+        }
+
+        private async Task<bool> TransferToParticipant(string targetParticipant)
+        {
+            transferToParticipantCompleteTask = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var identifierKind = GetIdentifierKind(targetParticipant);
+
+            if (identifierKind == CommunicationIdentifierKind.UnknownIdentity)
+            {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Unknown identity provided. Enter valid phone number or communication user id");
+                transferToParticipantCompleteTask.TrySetResult(true);
+            }
+            else
+            {
+                var operationContext = Guid.NewGuid().ToString();
+
+                RegisterToTransferParticipantsResultEvent(operationContext);
+
+                if (identifierKind == CommunicationIdentifierKind.UserIdentity)
+                {
+                    var response = await callConnection.TransferToParticipantAsync(new CommunicationUserIdentifier(targetParticipant), null, null, operationContext).ConfigureAwait(false);
+                    Logger.LogMessage(Logger.MessageType.INFORMATION, $"TransferParticipantAsync response --> {response.GetRawResponse()}, status: {response.Value.Status}  " +
+                        $"OperationContext: {response.Value.OperationContext}, OperationId: {response.Value.OperationId}, ResultDetails: {response.Value.ResultDetails}");
+                }
+                else if (identifierKind == CommunicationIdentifierKind.PhoneIdentity)
+                {
+                    var response = await callConnection.TransferToParticipantAsync(new PhoneNumberIdentifier(targetParticipant), null, null, operationContext).ConfigureAwait(false);
```

Comment:
dupe code here when calling PMA, the TransferToParticipantAsync() takes CommunicationIdentifier as the argument. so you don't need to create separate CommunicationUserIdentifier and PhoneNumberIdentifier.